src/Middleware/HealthChecks/src/HealthCheckMiddleware.cs

```diff
@@ -55,6 +55,13 @@ public async Task InvokeAsync(HttpContext httpContext)
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            if (!(HttpMethods.IsGet(httpContext.Request.Method) || HttpMethods.IsHead(httpContext.Request.Method)))
+            {
+                httpContext.Response.StatusCode = StatusCodes.Status405MethodNotAllowed;
+                httpContext.Response.Headers.Add(HeaderNames.Allow, new [] { HttpMethods.Get, HttpMethods.Head });
```

Comment:
Can `new [] { HttpMethods.Get, HttpMethods.Head }` be stored in a readonly static field?